src/utils/btc.ts

```diff
@@ -0,0 +1,83 @@
+import * as ecc from "@bitcoin-js/tiny-secp256k1-asmjs";
+import { initEccLib, address, networks } from "bitcoinjs-lib";
+import { NO_COORD_PK_BYTE_LENGTH } from "../constants/keys";
+
+// Initialize elliptic curve library
+export function initBTCCurve() {
+  initEccLib(ecc);
+}
+
+/**
+ * Check whether the given address is a valid Bitcoin address.
+ *
+ * @param {string} btcAddress - The Bitcoin address to check.
+ * @param {object} network - The Bitcoin network (e.g., bitcoin.networks.bitcoin).
+ * @returns {boolean} - True if the address is valid, otherwise false.
+ */
+export const isValidBitcoinAddress = (
+  btcAddress: string,
+  network: networks.Network,
+): boolean => {
+  try {
+    return !!address.toOutputScript(btcAddress, network);
```

Comment:
Why can't we use `address.fromBech32` here?

Reply:
>  i did however googled around, it seems this method works with any address type without the need of checking the prefix and version manually.

I thought you were saying that `address.fromBech32` is only useful if you want to check the prefix of the address like we do in `isTaproot`. What I was mentioning is that given that we use this in `isTaproot` we could use it here as well but without checking the prefixes.

Reply:
Sounds good